build/mongosh/test.js.example

```diff
@@ -0,0 +1,31 @@
+const assert = require('assert');
+
+(function() {
+    "use strict";
+    const coll = db.issue1539;
+
+    coll.drop();
+    
+    docs = [
+      { _id: "double", v: 42.13 },
+      { _id: "double-whole", v: 42.0 },
+      { _id: "double-zero", v: 0.0 },
+      { _id: "double-max", v: Number.MAX_VALUE },
+      { _id: "double-smallest", v: Number.MIN_VALUE },
+      { _id: "doubleBig", v: Number(2 << 60) },
+      { _id: "double-null", v: null }
+    ]
+
+    coll.insertMany(docs);
+
+    let noError = ""; // bad error
+    let expectedError = "Failed to parse $size. Expected a non-negative number in: $size: -1"; // good error
+
+    x = { v: { $size: -1 } };
+    try {
+        coll.findOne(x);
+    } catch(error) {
+        assert.equal(error.message, expectedError)
+    }
```

Comment:
Not tested code so it might not even compile, this is something I'm trying to check. That the error really was caught

```suggestion
    errCaught = false
    expectedErr = true

    x = { v: { $size: -1 } };
    try {
        coll.findOne(x);
    } catch(error) {
        errCaught = true
        assert.equal(error.message, expectedError)
    }
    assert.equal(errCaught, expectedErr)
```

Reply:
If we catch the error then it's not `null` and we can do something.

Reply:
But why we try to catch exception at all?

Reply:
Yes good question, and I think I have a solution. It may be because the shell uses async methods. It seems to be working with `assert.rejects`.

Reply:
Let's comment that in the code, so those who come across this code knows why are catching error exception :hugs:

Reply:
Will do @chilagrow.

Reply:
I am resolving this conversation as I believe explaining why we use the `try..catch` here is unnecessary, as discussed on the call. Please unresolve if I am mistaken.

Reply:
Please put on pause; I will take a look later

Reply:
ack